### PR TITLE
[Seedless-onboarding] - tests for useMPC hook

### PR DESCRIPTION
**Pull request by schmanu**
## What it solves
- Adds tests for the `useInitMPC` hook
- Connecting wallet after rehydration
- emitting correct events when switching chain
- initialization of MPC Core Kit

## How to test it
- `yarn test useMPC`

## Checklist
* [x] I've tested the branch on mobile 📱
* [x] I've documented how it affects the analytics (if at all) 📊
* [x] I've written a unit/e2e test for it (if applicable) 🧑‍💻

### CODE REVIEW

File: src/hooks/wallets/mpc/useMPC.ts [hunk 1]
```diff
@@ -43,10 +51,8 @@ export const useInitMPC = () => {
       .init()
       .then(() => {
         setStore(web3AuthCoreKit)
-
         // If rehydration was successful, connect to onboard
-        if (web3AuthCoreKit.status === COREKIT_STATUS.INITIALIZED) {
-          console.log('Logged in', web3AuthCoreKit)
+        if (web3AuthCoreKit.provider) {
```
Comment (iamacook) [inline src/hooks/wallets/mpc/useMPC.ts:55]: Nit: let's invert this to an early return.

File: src/hooks/wallets/mpc/__tests__/useMPC.test.ts [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+      listeners: (eventName: string) => {
+        if (eventName === 'chainChanged') {
+          return [mockChainChangedListener]
+        }
+      },
+    }
+
+    setMPCCoreKitInstance({
+      provider: mockProviderBefore,
+    } as any)
```
Comment (iamacook) [inline src/hooks/wallets/mpc/__tests__/useMPC.test.ts:203]: Can we get rid of this `any`?

Comment (schmanu) [reply src/hooks/wallets/mpc/__tests__/useMPC.test.ts:203]: I can only switch it for a `as unknown as Web3AuthMPCCoreKit` which is the same in this context.
I don't want to mock the entire object as it's quite complex and we only use this one field of it in the hook.

File: src/hooks/wallets/mpc/__tests__/useMPC.test.ts [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+        authentication: RPC_AUTHENTICATION.NO_AUTHENTICATION,
+        value: 'https://goerli.somerpc.io',
+      },
+    } as any)
+
+    const mockWeb3AuthMpcCoreKit = jest.spyOn(require('@web3auth/mpc-core-kit'), 'Web3AuthMPCCoreKit')
+    const mockChainChangedListener = jest.fn()
+    const mockProviderBefore = {
+      listeners: (eventName: string) => {
+        if (eventName === 'chainChanged') {
```
[the remark this answers is not in the document]
Comment (schmanu) [reply src/hooks/wallets/mpc/__tests__/useMPC.test.ts:195]: You cannot change your web3auth account. You only have one address.

File: src/hooks/wallets/mpc/__tests__/useMPC.test.ts [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+
+/**
+ * Small helper class that implements registering RPC event listeners and event emiting.
+ * Used to test that events onboard relies on are getting called correctly
+ */
+class EventEmittingMockProvider {
+  private chainChangedListeners: Function[] = []
+
+  addListener(event: string, listener: Function) {
+    if (event === 'chainChanged') {
```
[the remark this answers is not in the document]
Comment (schmanu) [reply src/hooks/wallets/mpc/__tests__/useMPC.test.ts:49]: You cannot change your web3auth account. You only have one address.

File: src/hooks/wallets/mpc/__tests__/useMPC.test.ts (hunk 2 above)
Comment (iamacook) [reply src/hooks/wallets/mpc/__tests__/useMPC.test.ts:203]: I'd suggest opting for that over `any`.

**Review by iamacook — APPROVED** [review]

**Pull request merged**